### PR TITLE
Add batch op - [RADAR-601]

### PR DESCRIPTION
**Pull request by junosuarez**
# Refereces
- [RADAR-601](https://zendesk.atlassian.net/browse/RADAR-601)
# Risks
- Low: adds new message type to protocol

### CODE REVIEW

File: lib/batch_message.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+function Batch () {
```
Comment (vanchi-zendesk) [inline lib/batch_message.js:1]: might want to take an array here already.

File: lib/batch_message.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	var messages = Array.prototype.slice.call(arguments)
+
+	this.op = 'batch'
+	this.value = messages
+}
+
+Object.defineProperty(Batch.prototype, 'length', {
```
Comment (vanchi-zendesk) [inline lib/batch_message.js:8]: is this supported on browsers we need? https://support.zendesk.com/hc/en-us/articles/203661786-Zendesk-system-requirements

**Comment by vanchi-zendesk** [comment]
:+1:

File: lib/batch_message.js (hunk 2 above)
Comment (junosuarez) [inline lib/batch_message.js:8]: yes, IE9 (the lowest common denominator) supports Object.defineProperty http://kangax.github.io/compat-table/es5/#ie9

File: lib/batch_message.js (hunk 1 above)
Comment (junosuarez) [inline lib/batch_message.js:1]: The api I chose was variadic to target ES6 ...spread (splat) notation

Comment (vanchi-zendesk) [inline lib/batch_message.js:1]: oh duh. didn't look at the next line closely enough. :/

**Pull request merged**